src/other/index.jl

```diff
@@ -214,6 +214,38 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
     getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
+
+@inline function Base.getindex(x::AbstractIndex, notidx::Union{Not{<:AbstractVector{Symbol}},
+                                                         Not{<:AbstractVector{<:AbstractString}}})
+    allunique(notidx.skip) || throw(ArgumentError("Elements of $(notidx.skip) must be unique"))
+    setdiff(1:length(x), [getindex(x, idx) for idx in notidx.skip if haskey(x, idx)])
+end
+
+@inline function Base.getindex(x::AbstractIndex, notidx::Not{<:AbstractVector{<:Integer}})
```

Comment:
why is this method needed? I would think it is covered by https://github.com/JuliaData/DataFrames.jl/pull/2228/files#diff-084f63e064c680ae660e2676e15d3d4eR249.

Then also https://github.com/JuliaData/DataFrames.jl/pull/2228/files#diff-084f63e064c680ae660e2676e15d3d4eR229 is not needed

Reply:
After you suggested we throw an error for repeated elements in the array I had to add a check that isn't present in the `Any` method. 

But then since `Bool <: Integer` I had to add another method because obviously `Vector{Bool}` will contain repeated elements. 

Let me know if that makes sense.

Reply:
My question is because we make this check in line https://github.com/JuliaData/DataFrames.jl/pull/2228/files#diff-084f63e064c680ae660e2676e15d3d4eR250. So now you would be doing the same check twice in this branch.

Now the check is needed in https://github.com/JuliaData/DataFrames.jl/pull/2228/files#diff-084f63e064c680ae660e2676e15d3d4eR220 because in line https://github.com/JuliaData/DataFrames.jl/pull/2228/files#diff-084f63e064c680ae660e2676e15d3d4eR221 you do not use a default `getindex` (which does this test) but a custom one (which does not do it).

Is this clear now what I meant? Thank you for working on this!